packages/react-button/src/components/Button/useButtonClasses.ts

```diff
@@ -1,489 +1,403 @@
-            [`.${GlobalClassNames.icon}`]: {
-              color: 'var(--button-highContrast-hovered-iconColor, --button-highContrast-iconColor)',
-            },
-          },
-        },
+    // TODO: spec calls out "shadow 2 __darker__", are we missing tokens?
+    shadowPressed: theme.alias.shadow.shadow2,
+  },
+  primaryDisabled: {
+    background: theme.alias.color.neutral.neutralBackgroundDisabled,
+    // borderColor: theme.alias.color.neutral.neutralStrokeDisabled,
+    color: theme.alias.color.neutral.neutralForegroundDisabled,
+    content2Color: theme.alias.color.neutral.neutralForegroundDisabled,
+    shadow: 'none',
+    shadowPressed: 'none',
+  },
+});
 
-        '&:active': {
-          background: 'var(--button-pressed-background, var(--button-hovered-background))',
-          color: 'var(--button-pressed-contentColor, var(--button-hovered-contentColor, var(--button-contentColor)))',
-          borderColor:
-            'var(--button-pressed-borderColor, var(--button-hovered-borderColor, var(--button-borderColor)))',
-          boxShadow: 'var(--button-pressed-boxShadow, var(--button-hovered-boxShadow, var(--button-boxShadow)))',
-
-          transform: 'var(--button-pressed-transform)',
-          transition: 'var(--button-pressed-transition)',
-
-          [`.${GlobalClassNames.icon}`]: {
-            color: 'var(--button-pressed-iconColor, var(--button-iconColor))',
-          },
-
-          [EdgeChromiumHighContrastSelector]: {
-            background:
-              'var(--button-highContrast-pressed-background, ' +
-              'var(--button-highContrast-hovered-background, ' +
-              'var(--button-highContrast-background)))',
-            borderColor:
-              'var(--button-highContrast-pressed-borderColor, ' +
-              'var(--button-highContrast-hovered-borderColor, ' +
-              'var(--button-highContrast-borderColor)))',
-            color:
-              'var(--button-highContrast-pressed-contentColor, ' +
-              'var(--button-highContrast-hovered-contentColor, ' +
-              'var(--button-highContarst-contentColor)))',
-            [`.${GlobalClassNames.icon}`]: {
-              color:
-                'var(--button-highContrast-pressed-iconColor, ' +
-                'var(--button-highContrast-hovered-iconColor, ' +
-                'var(--button-highContrast-iconColor)))',
-            },
-          },
+const useRootClasses = makeStyles<ButtonStyleSelectors>([
+  [
+    null,
+    theme => {
+      const buttonTokens = makeButtonTokens(theme);
+
+      return {
+        ':global(:root)': {
+          // TODO: how to define and access component variables that do not have globals/aliases?
+          //       consider IE11 friendly way
+          '--button-height': '32px',
+          '--button-paddingX': '12px',
+          '--button-minWidth': '96px',
+          '--button-maxWidth': '280px',
+
+          '--button-iconOnly-padding': '6px',
```

Comment:
I don't see any usage of these variables, cruft?

Reply:
Removed, thanks. We'll address component variables once this PR is merged and we look holistically at the converged components we have.